app/scripts/controllers/detect-tokens.js

```diff
@@ -200,13 +245,17 @@ export default class DetectTokensController {
    * @type {number}
    */
   set interval(interval) {
-    this._handle && clearInterval(this._handle);
     if (!interval) {
       return;
     }
-    this._handle = setInterval(() => {
-      this.detectNewTokens();
-    }, interval);
+    if (isManifestV3) {
+      this.rescheduleTokenDetectionPollingInMV3();
+    } else {
+      this._handle && clearInterval(this._handle);
```

Comment:
Nit:  I think we've tried getting away from the `&&` pattern and gone with `if`'s.